lib/navigation/imu_displacement.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include "core/time.hpp"
+#include "core/timer.hpp"
+#include "core/types.hpp"
+
+namespace hyped::navigation {
+
+class ImuDisplacement {
+ public:
+  ImuDisplacement(const core::ITimeSource &time, const core::TimePoint initial_time);
+  /**
+   * @brief update the imu estimate for both displacement and velocity. Displacement is
+   * used for cross checking agreement with wheel encooders, velocity is used for us to
+   * estimate velocity in the case we have no direct velocity sensors.
+   *
+   * @param imu_acceleration kalman filtered single value for acceleration
+   * @param imu_timestamp time at which easurement was taken
+   */
+  void updateImuDisplacement(const core::Float imu_acceleration,
+                             const core::TimePoint imu_timestamp);
+
+  core::Float getImuDisplacement();
+
+  core::Float getImuVelocity();
+
+ private:
+  core::Float imu_displacement_;
```

Comment:
`displacement_estimate_`?

Reply:
I just wanted to make very clear that this is the imu estimate for displacement since we will also have encoder and keyence estimates

Reply:
but you're inside the IMU related class? I also want to emphasise that this is an estimate.